### PR TITLE
fibonecci

### PR DESCRIPTION
**Pull request by TomaszWeglinski2**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: fibonacci
🏅 Granted 6 XP for @TomaszWeglinski2

**Comment by ziobron** [comment]
⏱ 3 XP extra for @TomaszWeglinski2 for on-time delivery

File: homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,27 @@
 #pragma once
 
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int a=0;
+    int b=1;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:5]: Add space between operations like '+' '-' '!=' '==' etc...

Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:5]: What are 'a' and `b` ? Variable should always represent real operation.
Like `result` `divider` `multiplier` `error_code` etc...

File: homework/fibonacci/fibonacci.hpp [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+
+    if(sequence==0){
+        return 0;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:9]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 3]
```diff
@@ -1,11 +1,27 @@
...
+    for (int i =0; i<sequence-1;i++){
+
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:11]: redundant new line

File: homework/fibonacci/fibonacci.hpp [hunk 4]
```diff
@@ -1,11 +1,27 @@
...
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    int a=0;
+    int b=1;
+
+    if(sequence==0){
+        return 0;
+    }
+    for (int i =0; i<sequence-1;i++){
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:10]: spaces

File: homework/fibonacci/fibonacci.hpp [hunk 5]
```diff
@@ -1,11 +1,27 @@
...
+
+        b+=a;
+        a=b-a;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:13]: spaces

File: homework/fibonacci/fibonacci.hpp [hunk 6]
```diff
@@ -1,11 +1,27 @@
...
 
 int fibonacci_recursive(int sequence) {
-    // TODO: Your implementation goes here
-    return 0;
+    if (sequence ==0){
+        return 0;
+    }
+    if(sequence ==1){
+        return 1;
+    }
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:24]: Sequence hold exactly the same value as you want to return for 0 and 1 :)

Better write:

`if (sequence < 2) {
    return sequence;
}`

File: homework/fibonacci/fibonacci.hpp [hunk 7]
```diff
@@ -1,11 +1,27 @@
...
+
+    return fibonacci_recursive(sequence-2) + fibonacci_recursive(sequence-1) ;
```
Comment (nauka-programowania-MA) [inline homework/fibonacci/fibonacci.hpp:26]: Add space between operations like '+' '-' '!=' '==' etc...

**Pull request closed without merging**